htdocs/core/class/notify.class.php

```diff
@@ -442,122 +443,149 @@ public function send($notifcode, $object, $filename_list = array(), $mimetype_li
 
 						switch ($notifcode) {
 							case 'BILL_VALIDATE':
-								$link = '<a href="'.$urlwithroot.'/compta/facture/card.php?facid='.$object->id.'">'.$newref.'</a>';
-								$dir_output = $conf->facture->dir_output;
+								$link = '<a href="'.$urlwithroot.'/compta/facture/card.php?facid='.$object->id.'&entity='.$object->entity.'">'.$newref.'</a>';
```

Comment:
Why adding entity into url ? Entity should be found from the object id itself and providing on url is not reliable.

Reply:
because with multicompany and not fullsharing mode; basic notification send link but always to object in first entity and display error.